runtime/frequency/src/lib.rs

```diff
@@ -1268,279 +1254,6 @@ construct_runtime!(
-					state_owner_msa_id,
-					payload.schema_id,
-					payload.page_id,
-					payload.target_hash,
-				)],
-				_ => vec![],
-			},
-			RuntimeCall::StatefulStorage(StatefulStorageCall::delete_page_with_signature_v2 {
-				payload,
-				delegator_key,
-				..
-			}) => match Msa::ensure_valid_msa_key(delegator_key) {
-				Ok(state_owner_msa_id) => vec![HashCheckData::new_paginated(
-					state_owner_msa_id,
-					payload.schema_id,
-					payload.page_id,
-					payload.target_hash,
-				)],
-				_ => vec![],
-			},
-			RuntimeCall::FrequencyTxPayment(FrequencyPaymentCall::pay_with_capacity {
-				call,
-				..
-			}) => Self::extract_hash_data(call),
-			RuntimeCall::FrequencyTxPayment(
-				FrequencyPaymentCall::pay_with_capacity_batch_all { calls, .. },
-			) => calls.iter().flat_map(|c| Self::extract_hash_data(c)).collect(),
-			RuntimeCall::Utility(UtilityCall::batch { calls, .. }) |
-			RuntimeCall::Utility(UtilityCall::batch_all { calls, .. }) =>
-				calls.iter().flat_map(|c| Self::extract_hash_data(c)).collect(),
-			_ => vec![],
-		}
-	}
-
-	/// Verifies the hashes for an Itemized Stateful Storage extrinsic
-	fn verify_hash_itemized(
-		msa_id: &MessageSourceId,
-		schema_id: &SchemaId,
-		target_hash: &PageHash,
-	) -> TransactionValidity {
-		const TAG_PREFIX: &str = "StatefulStorageHashItemized";
-
-		if let Ok(Some(page)) = StatefulStorage::get_itemized_page_for(*msa_id, *schema_id) {
-			let current_hash: PageHash = page.get_hash();
-			ensure!(
-				&current_hash == target_hash,
-				Self::map_dispatch_error(
-					pallet_stateful_storage::Error::<Runtime>::StalePageState.into()
-				)
-			);
-
-			return ValidTransaction::with_tag_prefix(TAG_PREFIX)
-				.and_provides((msa_id, schema_id))
-				.build();
-		}
-		Ok(Default::default())
-	}
-
-	/// Verifies the hashes for a Paginated Stateful Storage extrinsic
-	fn verify_hash_paginated(
-		msa_id: &MessageSourceId,
-		schema_id: &SchemaId,
-		page_id: &PageId,
-		target_hash: &PageHash,
-	) -> TransactionValidity {
-		const TAG_PREFIX: &str = "StatefulStorageHashPaginated";
-		if let Ok(Some(page)) =
-			StatefulStorage::get_paginated_page_for(*msa_id, *schema_id, *page_id)
-		{
-			let current_hash: PageHash = page.get_hash();
-
-			ensure!(
-				&current_hash == target_hash,
-				Self::map_dispatch_error(
-					pallet_stateful_storage::Error::<Runtime>::StalePageState.into()
-				)
-			);
-
-			return ValidTransaction::with_tag_prefix(TAG_PREFIX)
-				.and_provides((msa_id, schema_id, page_id))
-				.build();
-		}
-
-		Ok(Default::default())
-	}
-
-	/// Map a module DispatchError to an InvalidTransaction::Custom error
-	fn map_dispatch_error(err: DispatchError) -> InvalidTransaction {
-		InvalidTransaction::Custom(match err {
-			DispatchError::Module(module_err) =>
-				<u32 as Decode>::decode(&mut module_err.error.as_slice())
-					.unwrap_or_default()
-					.try_into()
-					.unwrap_or_default(),
-			_ => 255u8,
-		})
-	}
-}
-
```

Comment:
nit: assuming there is still stalehash check in extrinsics?

Reply:
That is correct, we didn't remove that check.